dGame/dComponents/ModelComponent.cpp

```diff
@@ -116,8 +116,11 @@ void ModelComponent::Serialize(RakNet::BitStream& outBitStream, bool bIsInitialU
 	if (bIsInitialUpdate) outBitStream.Write0(); // We are not writing model editing info
 }
 
-void ModelComponent::UpdatePendingBehaviorId(const int32_t newId) {
-	for (auto& behavior : m_Behaviors) if (behavior.GetBehaviorId() == -1) behavior.SetBehaviorId(newId);
+void ModelComponent::UpdatePendingBehaviorId(const LWOOBJID newId, const LWOOBJID oldId) {
+	for (auto& behavior : m_Behaviors) {
+		if (behavior.GetBehaviorId() != oldId) continue;
+		behavior.SetBehaviorId(newId);
+	}
```

Comment:
This is the only functional change in this pr. From the more recent one and the corresponding call changes are at the top of ControlBehaviors.cpp